### ISSUE TITLE
Support dynamic paths in VRL

### ISSUE THREAD

**Issue by jszwedko** [vectordotdev/vector#7050]
Requested by user in discord: https://discord.com/channels/742820443487993987/746070591097798688/829794156062310479

They'd like to be able to lookup a path on an object by a variable, for example something like:

```
name = "Timmy"
ages = {"Timmy": 10}
lookup(ages, name) # or maybe ages.$name or ages[name]
```

**Comment by JeanMertz** [comment]
I think the best first step is to add a `lookup` function. We can offer dedicated syntax in the future, if there's a clear need.

**Comment by JeanMertz** [comment]
We should also add `insert` alongside `lookup`:

```coffee
. = { "foo": "bar", "bar": true }

.baz = lookup!(., .foo)
// { "foo": "bar", "bar": true, "baz": true }

. = insert!(., upcase!(.foo) + "BAZ", true)
// { "foo": "bar", "bar": true, "baz": true, "BARBAZ": true }

**Comment by ypid-geberit** [comment]
`ages[name]` would be my first choice because then you don’t even have to look into the docs. It just works as you are used to from other languages.

**Comment by JeanMertz** [comment]
> `ages[name]` would be my first choice because then you don’t even have to look into the docs. It just works as you are used to from other languages.

That's definitely something we want to work towards 👍

As a first step, we'll be adding functions to do this, since they are easier to add, and unblock this specific restriction without having to change the VRL parser.

**Comment by StephenWakely** [comment]
This has been addressed by #9284.
